hw/top_earlgrey/data/ip/chip_uart_testplan.hjson

```diff
@@ -0,0 +1,252 @@
+                 - Set FIFO_CTRL.RXILVL to 1 so that one character can be received
+                 - Enable RX
+                 - Transfer a single packet from testbench to UART IO with parity bit included
+                 - Check if there is any interrupt and if the source is from UART
+                 - Check if FIFO_STATUS.RXLVL is 1
+                 - Clear the rx_watermark interrupt
+                 - Repeat the sequence with 10 packets in a single transaction
+
+               Transmit sequence:
+                 - Set FIFO_CTRL.TXILVL to 1 so that one character can be transmitted
+                 - Write a random data to WDATA register
+                 - Enable TX
+                 - Monitor UART IO for transmission of data packet
+                 - Check if the packet framing and parity is correct
+                 - Check if there is any interrupt and if the source is from UART
+                 - Check if there is tx_empty interrupt after transmission of one packet
+                 - Clear the rx_empty interrupt
+                 - Repeat the sequence with 10 packets in a single transaction
+            '''
+      stage: V1
+      si_stage: SV1
+      lc_states: ["TEST_UNLOCKED", "PROD"]
+      features: ["UART.RX_EN", "UART.TX_EN", "UART.RX_WATERMARK_INTERRUPT",
+                 "UART.PARITY_ODD", "UART.PARITY_EN", "UART.BAUD_RATE_CONTROL"]
+      tests: ["chip_sw_uart_tx_rx_parity_smoke", "chip_sw_uart_tx_rx"]
+    },
+    // Feature tests
+    {
+      name: chip_sw_uart_loopback
+      desc: ''' Verify the loopback feature of UART
+
+                - Powerup OT and enable UART IO
+                - Calculate CONTROL.NCO using programmers model and update the register
+                - Enable transmission and reception by setting CONTROL.RX and CONTROL.TX
+                - Transmit random bit pattern from testbench to UART
+                - After transmitting each bit wait for one clock period and check the received bit on UART IO
+                - Check if the bit is same as transmitted in earler cycle
+            '''
+      stages: V2
+      si_stage: SV2
+      lc_states: ["TEST_UNLOCKED", "PROD"]
+      features: ["UART.RX_EN", "UART.TX_EN", "UART.LOOPBACK", "UART.TX_WATERMARK_INTERRUPT", "UART.BAUD_RATE_CONTROL"]
+      tests: ["chip_sw_uart_loopback"]
+    },
+    {
+      name: chip_sw_uart_sloopback
+      desc: ''' Verify the system loopback feature of UART
+
+                - Powerup OT and enable UART IO
+                - Calculate CONTROL.NCO using programmers model and update the register
+                - Write 10 random bytes in to WDATA FIFO
+                - Enable TX and RX
+                - Set FIFO_CTRL.TXLVL to 1 so that ibex gets notified for each byte transmitted
+                - In ibex, Wait for interrupt from UART
+                - If tx_watermark interrupt is asserted
+                  - Read data from RDATA FIFO and check if is the same written to WDATA FIFO
```

Comment:
I think this should be "... is the same as was written .." ?

Reply:
updated description

Reply:
I think the "as was" is also worth including. Otherwise it can sound like it's saying "The data may not be the same yet but it will look the same once it has been written" (not what you mean!)